src/rfc3986/misc.py

```diff
@@ -51,7 +51,7 @@
     (
         "^(?:(?P<userinfo>{})@)?"  # userinfo
         "(?P<host>{})"  # host
-        ":?(?P<port>{})?$"  # port
+        "(:(?P<port>{}))?$"  # port
```

Comment:
Should be a non-capturing group on the outside as we don't need to capture `:<port>` as well as `<port>`

Reply:
Yup, good catch. Fixed it just now.